src/simulation/power/simpleBattery/simpleBattery.cpp

```diff
@@ -59,6 +59,16 @@ void SimpleBattery::evaluateBatteryModel(PowerStorageStatusMsgPayload *msg) {
         this->storedCharge = 0;
     }
 
+    if(this->fault == true) {   
```

Comment:
Is there some value to making these `fault` and `faultCapacityRatio` input messages instead of class variables?

Reply:
Mm, you are thinking like a battery configuration message?  I'm not sure I can think of an associated scenario that would need this.  I'll think on this.

Reply:
Mm, interesting.  I could see value in that.  @Yume27 , what you can do is move the fault state to a message.  Then, in `simpleBattery` you check if this optional message is connected.  If yes, you read in the values and use them?  In your event handler you would just change the value of this input message.

Reply:
Also, couldn't this fault feature be added to the `powerStorageBase()` base class?  This way all batteries based on this parent class would get this capability?